climada/engine/impact.py

```diff
@@ -1389,6 +1389,184 @@ def _selected_exposures_idx(self, coord_exp):
+            return unit
+
+        def check_crs(imp_list):
+            crs = [imp.crs for imp in imp_list if imp.crs != '']
+            if len(crs) < len(imp_list):
+                raise ValueError(
+                    "Not all impacts have a CRS. Specify the same CRS for each impact."
+                    "The impacts are incompatible and cannot be concatenated.")
+            if len(set(crs)) > 1:
+                raise ValueError(f"The given impacts have different crs: {crs}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            return crs[0]
+
+        def check_attributes(imp_list):
+            attributes = set.union(*[set(vars(imp).keys()) for imp in imp_list])
+            for attr_name in attributes:
+                if not all(hasattr(imp, attr_name) for imp in imp_list):
+                    raise ValueError(f"Attribute {attr_name} is not shared by all impacts. "
+                                    "The impacts are incompatible and cannot be concatenated.")
+            return True
+
+        def check_dates(imp_list):
+
+            #compare dates of all impact pairs
+            for i, imp1 in enumerate(imp_list):
+                for imp2 in imp_list[i+1::]:
+                    dates_a_set = set(imp1.date)
+                    dates_b_set = set(imp2.date)
+                    if len(dates_a_set.intersection(dates_b_set)) > 0:
+                        raise ValueError("Found at least one duplicate date. "
+                                         "Impacts cannot contain same dates."
+                                         "Impacts are incompatible and cannot be concatenated.")
+            date = np.array([date for imp in imp_list for date in imp.date])
+            return date
+
+        def check_exposure(imp_list):
+
+            #check total value
+            tot_vals = {imp.tot_value for imp in imp_list}
+            if len(tot_vals) > 1:
+                raise ValueError("The impacts are not based on the same exposure."
+                                 "The total exposure values are different."
+                                 "The impacts are incompatible and cannot be concatenated.")
+
+            #compare exposure coordinates of all impact pairs
+            for i, imp1 in enumerate(imp_list):
+                for imp2 in imp_list[i + 1::]:
+                    if not np.array_equal(imp1.coord_exp, imp2.coord_exp):
+                        raise ValueError("The impacts are not based on same exposure."
+                                         "Impacts are incompatible and cannot be concatenated.")
+            #compare exposure descriptions
+            exp_desc = {imp.tag['exp'].description for imp in imp_list
+                        if imp.tag['exp'].description != ''}
+            if len(exp_desc) > 1:
+                LOGGER.warning("Impacts are based on exposures with different descriptions."
+                               "Concatenate anyways.")
```

Comment:
Shouldn't the user be aware of that? I don't see how this information would help anybody 😅